### ISSUE TITLE
Support a prefix to filter autoupdate tags

### ISSUE THREAD

**Issue by JonZeolla** [pre-commit/pre-commit#3326]
### search you tried in the issue tracker

'autoupdate tags', 'autoupdate pre-release', etc.

## TL;DR

Similar prior issues:

- https://github.com/pre-commit/pre-commit/issues/3052
- https://github.com/pre-commit/pre-commit/issues/2988
- https://github.com/pre-commit/pre-commit/issues/2311
- https://github.com/pre-commit/pre-commit/issues/1925

... and many more. What I discuss below is backwards compatible, generic, and I will be opening a PR shortly to implement it.

If accepted I'd be happy to PR updates to the [docs](https://github.com/pre-commit/pre-commit.com/blob/28b91e42d01e8a5f8c7467f97b6d721ab9625955/sections/cli.md?plain=1#L14) as well.

## Actual problem

I want to constrain my `pre-commit autoupdate`s to a tag prefix or some sort of match logic, perhaps like `pre-commit autoupdate --freeze --prefix v`, `--tag-prefix v`, or to more closely map to the actual git commands being used, `--match "v*"` (with the obvious usability downside of an unquoted `*` causing mistakes/questions).

Some projects regularly push/update tags which may not be desirable, such as `lychee`'s [`nightly` tag](https://github.com/lycheeverse/lychee/tree/nightly) (which is how I encountered this), but many others also note alpha/beta/pre-release tags causing similar issues.

## Potential solution

I envision something like:

```bash
pre-commit autoupdate --tags-prefix v --repo https://github.com/asottile/reorder-python-imports
```

After a brief look at [the update code](https://github.com/pre-commit/pre-commit/blob/cc4a52241565440ce200666799eef70626457488/pre_commit/commands/autoupdate.py#L38) it seems one way to implement this may be by using the `git` argument of `--match`:

```bash
$ git -c core.useBuiltinFSMonitor=false -C ./lychee describe FETCH_HEAD --tags --abbrev=0
nightly
$ git -c core.useBuiltinFSMonitor=false -C ./lychee describe FETCH_HEAD --tags --abbrev=0 --match "v*"
v0.15.1
```

This would not work if `tags_only` is `False`:

```bash
$ git -c core.useBuiltinFSMonitor=false -C ./lychee describe FETCH_HEAD --tags --exact --match "v*"
fatal: no tag exactly matches '18d486d782330b096afdabaf82950b1636c9f021'
```

However, it seems like that [comes from `--bleeding-edge`](https://github.com/pre-commit/pre-commit/blob/cc4a52241565440ce200666799eef70626457488/pre_commit/main.py#L379) which would be incompatible with a tag prefix, so those arguments would be mutually exclusive.

## Background

I automate PRs that update my dependencies (including these pre-commit revs) to simplify maintenance, but if I want to avoid a tag like `nightly` it currently requires manual changes to revert that `rev` or find a new `v*` tag that the project has published.

### pre-commit --version

pre-commit 4.0.1

**Comment by asottile** [comment]
this has been rejected many times as you've found.  please take no as an answer
